.github/workflows/CI.yml

```diff
@@ -258,12 +269,19 @@ jobs:
         uses: actions/cache@v2
         with:
           key: "cached_pkg_${{ github.sha }}"
-          path: go.mod
+          path: |
+            %LocalAppData%\go-build
+            ~/go/pkg/mod
 
+      - name: Unit Test 
+        if: steps.cached_pkg.outputs.cache-hit != 'true'
+        working-directory: cwa
+        run: make test
```

Comment:
Adding `make test` for unit test Linux and MacOs in their respective environment before distributing the binaries to MacOS, Windows, Linux, Darwins